src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -348,6 +348,8 @@ private static bool IsPossibleStartOfTypeDeclaration(SyntaxKind kind)
                 case SyntaxKind.SealedKeyword:
                 case SyntaxKind.StaticKeyword:
                 case SyntaxKind.UnsafeKeyword:
+                // PROTOTYPE(ft): it seems strange that we don't need this for any of the tests so far to pass.
```

Comment:
Add a test for `partial file Type`, and verify that we get a `partial` misplaced error.

Reply:
Or `async file void M()`

Reply:
It looks like this method is never called with the token's ContextualKind, only with the Kind. The parse we're giving for 'partial file class' is currently similar to the one we give for 'partial async class'. I think it would be reasonable to try and adjust the parsing code in a separate PR to give a better parse in both scenarios. I'll delete this line.